# Undertanding the Iris data set

The notebook aims to undertand the content of the Iris data set.


## Acknowledgments

- Data from https://www.coursera.org/ from the course "Understanding and Visualizing Data with Python" by University of Michigan


# Iris data set

# Importing and inspecting the data

In [16]:
# Import the packages that we will be using
import pandas as pd
import seaborn as sns

# Dataset url
ruta = "datasets/iris/irisFinal.csv"

# Load the dataset
df = pd.read_csv(ruta)

In [25]:
# Número de filas y columnas
NF, NC = df.shape
print("Número de filas:", NF, "\nNúmero de columnas:", NC)
print("")

# Número de observaciones por clase.
print(df.Class.value_counts())
print("")

# Tipos de variables
print(df.dtypes)
print("")

# Impresión de la tabla
print(df)

Número de filas: 150 
Número de columnas: 5

Iris-versicolor    50
Iris-virginica     50
Iris-setosa        50
Name: Class, dtype: int64

SepalLength    float64
SepalWidth     float64
PetalLength    float64
PetalWidth     float64
Class           object
dtype: object

       SepalLength  SepalWidth  PetalLength  PetalWidth
count   150.000000  150.000000   150.000000  150.000000
mean      5.843333    3.057333     3.758000    1.199333
std       0.828066    0.435866     1.765298    0.762238
min       4.300000    2.000000     1.000000    0.100000
25%       5.100000    2.800000     1.600000    0.300000
50%       5.800000    3.000000     4.350000    1.300000
75%       6.400000    3.300000     5.100000    1.800000
max       7.900000    4.400000     6.900000    2.500000


# Final remarks


- The **Pandas** library provides fancy, high-performance, easy-to-use data structures and data analysis tools


# Activity: work with the iris dataset

1. Load the iris.csv file in your computer and compare with the data printed above.
Tiene menos atributos que el dataset de Cartwheel, pero tiene más observaciones o más ejemplos/objetos.

2. How many observations (rows) are in total?
150

3. How many variables (columns) are in total? What do they represent?
5

4. How many observatoins are for each type of flower? 
50 por cada clase de flor.

5. What is the type of data for each variable?
SepalLength    float64
SepalWidth     float64
PetalLength    float64
PetalWidth     float64
Class           object

6. What are the units of each variable?
Sepal and Pedal Width and lenght: cm
Class: 3 types = Iris Setosa, Iris Versicolor, Iris virginica